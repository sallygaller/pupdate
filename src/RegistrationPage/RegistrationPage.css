.RegistrationPage h2 {
    text-align: center;
}

.RegistrationPage {
    border-radius: 5px;
    padding: 20px;  
}


