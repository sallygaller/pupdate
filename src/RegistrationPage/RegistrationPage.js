import React from "react";
import { withRouter } from "react-router-dom";
import RegistrationForm from "../RegistrationForm/RegistrationForm";
import "./RegistrationPage.css";

class RegistrationPage extends React.Component {
  static defaultProps = {
    history: {
      push: () => {},
    },
  };

  handleRegistrationSuccess = () => {
    this.props.history.push("/login");
  };

  render() {
    return (
      <div className="RegistrationPage">
        <h2>Welcome!</h2>
        <RegistrationForm
          onRegistrationSuccess={this.handleRegistrationSuccess}
        />
      </div>
    );
  }
}

export default withRouter(RegistrationPage);
