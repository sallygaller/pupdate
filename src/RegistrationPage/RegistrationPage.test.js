import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import RegistrationPage from "./RegistrationPage";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <RegistrationPage />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <RegistrationPage />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
