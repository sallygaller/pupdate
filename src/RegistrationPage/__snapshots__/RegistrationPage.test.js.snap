// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="RegistrationPage"
>
  <h2>
    Welcome!
  </h2>
  <form
    className="RegistrationForm"
    onSubmit={[Function]}
  >
    <div
      className="RegistrationForm-error"
      role="alert"
    />
    <label
      htmlFor="first-name"
    >
      First Name: 
      <span>
        *
      </span>
    </label>
    <input
      id="first-name"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <label
      htmlFor="last-name"
    >
      Last Name: 
      <span>
        *
      </span>
       
    </label>
    <input
      id="last-name"
      name="last-name"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <label
      htmlFor="email"
    >
      Email: 
      <span>
        *
      </span>
    </label>
    <input
      id="email"
      name="email"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <label
      htmlFor="password"
    >
      Password: 
      <span>
        *
      </span>
    </label>
    <p>
      (must be longer than 8 characters and contain 1 upper case, lower case, number and special character)
    </p>
    <input
      id="password"
      name="password"
      onChange={[Function]}
      required={true}
      type="password"
      value=""
    />
    <label
      htmlFor="zip"
    >
      City, State, Country: 
      <span>
        *
      </span>
    </label>
     
    <div
      className="search-location-input"
    >
      <input
        onChange={[Function]}
        placeholder="Portland, OR, USA"
        value=""
      />
    </div>
    <button
      type="submit"
    >
      Sign Up!
    </button>
  </form>
</div>
`;
